### PR TITLE
[fix bug 1256241] Implement /new download page redesign tests

### PR DESCRIPTION
**Pull request by alexgibson**
## Description
`/firefox/new/` redesign tests. These tests are to be conducted in Optimizely for en-US only, targeting non-Firefox browsers.

This is still a WIP, but opening up for general review as things are nearly there.

Scene1
-----------

- `/firefox/new/?v=1a`
- `/firefox/new/?v=2a`
- `/firefox/new/?v=3a`
- `/firefox/new/?v=4a`
- `/firefox/new/?v=1b`
- `/firefox/new/?v=2b`
- `/firefox/new/?v=3b`
- `/firefox/new/?v=4b`

Scene2
----------

- `/firefox/new/?scene=2&v=a`
- `/firefox/new/?scene=2&v=b`

## Bugzilla
https://bugzilla.mozilla.org/show_bug.cgi?id=1256241

## Testing
Mostly design changes (JS logic for the two pages remains the same). Designs should degrade gracefully in old browsers (text visible, download button visible).

Demo sever links: https://bugzilla.mozilla.org/show_bug.cgi?id=1256241#c11

## Checklist
- [ ] Requires l10n changes.
- [x] Related functional & integration tests passing.
- [x] Write unit tests for testing template logic.

### CODE REVIEW

**Comment by alexgibson** [comment]
Added a todo item to the description - I still need to write some unit tests for the testing template logic ^

**Comment by alexgibson** [comment]
Added some tests for the view logic :+1:

File: media/js/firefox/new/redesign/set-scene2-b.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+(function($) {
+    'use strict';
+
+    $('.download-link').each(function(i, link) {
+        if (link.href.indexOf('scene=2') > -1) {
+            // specify version a template
```
Comment (jpetto) [inline media/js/firefox/new/redesign/set-scene2-b.js]: Not that it really matters, but version b... :trollface:

File: media/css/firefox/new/redesign/scene2-common.less [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+    a {
+        display: block;
+        margin-bottom: @baseLine;
+    }
+}
+
+/* Tablet Layout: 760px */
+@media only screen and (min-width: @breakTablet) and (max-width: @breakDesktop) {
+
+}
```
Comment (jpetto) [inline media/css/firefox/new/redesign/scene2-common.less]: Plan for this MQ? Remove if not?

File: media/css/firefox/new/redesign/scene2-common.less [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+
+    .ground {
+        width: 100%;
+        height: 120px;
+        background: #11336B;
+    }
+}
+
+// If browser supports CSS blend modes, add grit texture over CSS gradient & apply color-dodge.
+@supports (background-blend-mode: color-dodge) {
```
Comment (jpetto) [inline media/css/firefox/new/redesign/scene2-common.less]: Yes! :star2:

File: media/css/firefox/new/redesign/scene1-browser.less [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+    @media only screen and (max-width: @breakDesktop) {
+        height: 98px;
+        .background-size(@widthTablet - (@gridGutterWidth * 2), auto);
+    }
+
+    @media only screen and (max-width: @breakTablet) {
+        height: 60px;
+        .background-size(@widthMobileLandscape, auto);
+    }
```
Comment (jpetto) [inline media/css/firefox/new/redesign/scene1-browser.less]: The browser image becomes borderline noise at this breakpoint, and is pretty useless at `@breakMobileLandscape`. Probably not a very common condition (desktop with skinny screen), but maybe hide the image at either `@breakTablet` or `@breakMobileLandscape` for versions 2 and 4? The variations would lose an important distinguishing feature (that I don't think we'd ship as a final), but would still be unique.

What do you think?

File: media/css/firefox/new/redesign/common.less [hunk 5]
```diff
@@ -0,0 +1,232 @@
...
+    }
+
+    @media only screen and (max-width: @breakDesktop) {
+        &:before {
+            display: none;
+        }
+    }
+}
+
+h1, h2, h3, h4, h5, h6, .large {
```
Comment (jpetto) [inline media/css/firefox/new/redesign/common.less]: Could put these on separate lines for consistency if making other updates.

**Comment by jpetto** [comment]
Looking really good! Tested in IE 6 - Edge, Safari, Chrome, & Fx. A couple very small notes, but nothing at all major.

If I had to make one UI suggestion (if time allows), it would be to make the twinkling just a *little* more noticeable. I actually missed it during testing until I saw animation calls in the CSS. :grin:

File: media/css/firefox/new/redesign/scene2-common.less (hunk 3 above)
Comment (alexgibson) [inline media/css/firefox/new/redesign/scene2-common.less]: I know right!? :bowtie:

File: media/css/firefox/new/redesign/scene1-browser.less (hunk 4 above)
Comment (alexgibson) [inline media/css/firefox/new/redesign/scene1-browser.less]: Yeah I think this makes sense, I don't think small viewports would really lose much if we hid the browser graphic here.

**Comment by alexgibson** [comment]
Thanks for the review @jpetto - updated (and added a little faster twinkle to the stars). Once we have a final design chosen, I'd love to revisit and spend a little more time on the animations. But between this and the Yahoo! work this sprint, I'm pressed to have as much time to spend on this as I'd like.

**Comment by alexgibson** [comment]
Adding a do-not-merge label for now, as it seems we're still waiting on GA (unclear if anything else is needed, or if we're good to go).

**Comment by jpetto** [comment]
Looking good to me. I'll leave an r+ here for whenever we're ready to merge.

**Comment by alexgibson** [comment]
Looks like we have the green light to merge this :bulb: 

https://bugzilla.mozilla.org/show_bug.cgi?id=1256241#c21

**Comment by alexgibson** [comment]
Spoke too soon, may need to remove the footer.

**Comment by alexgibson** [comment]
After a whirlwind of discussion on IRC we settled on the footer not needing to change for this particular test.

**Comment by alexgibson** [comment]
@jpetto unless there's a reason not to (which I may not be aware of), could you please merge this on Monday? I think Chris would like to get the Optimizely experiments set up once this is in production. Thanks!

**Comment by jpetto** [comment]
@alexgibson - Yep, I mentioned to cmore in the sprint presentation on Friday that we'd merge this on Monday and get the tests started. :+1:

**Pull request merged**